[cutlass backend] cache a few things for codegen and properties

henrylhtsang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #158158

Differential Revision: [D78193404](https://our.internmc.facebook.com/intern/diff/D78193404/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang committed:
[cutlass backend] cache a few things for codegen and properties



Differential Revision: [D78193404](https://our.internmc.facebook.com/intern/diff/D78193404/)

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] cache a few things for codegen and properties"


Differential Revision: [D78193404](https://our.internmc.facebook.com/intern/diff/D78193404/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] cache a few things for codegen and properties"


Differential Revision: [D78193404](https://our.internmc.facebook.com/intern/diff/D78193404/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler commented on torch/_inductor/codegen/cuda/cuda_kernel.py:
```diff
@@ -627,6 +628,7 @@ def kernel_hash_key(self) -> str:
             ]
         )
 
+    @functools.cache  # noqa: B019
```
ditto, it doesn't sound like join would be time consuming. And how often is this ran?

ColinPeppler commented on torch/_inductor/codecache.py:
```diff
@@ -3708,6 +3708,7 @@ def get_kernel_binary_remote_cache(
             return None
 
     @classmethod
+    @lru_cache(None)
```
how much is caching this helping? write is supposed to do caching.

henrylhtsang committed:
Update on "[cutlass backend] cache a few things for codegen and properties"


Differential Revision: [D78193404](https://our.internmc.facebook.com/intern/diff/D78193404/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] cache a few things for codegen and properties"


Differential Revision: [D78193404](https://our.internmc.facebook.com/intern/diff/D78193404/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] cache a few things for codegen and properties"


Differential Revision: [D78193404](https://our.internmc.facebook.com/intern/diff/D78193404/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] cache a few things for codegen and properties"


Differential Revision: [D78193404](https://our.internmc.facebook.com/intern/diff/D78193404/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]
